### PR TITLE
Check for oscap RPMs before trying to run a scan

### PR DESCRIPTION
**Pull request by akofink**
- Handle the case when the SCAP rpms are not installed.
- Move to using insights.util.subproc

Fixes https://projects.engineering.redhat.com/browse/RHICOMPL-359, https://projects.engineering.redhat.com/browse/RHICOMPL-360

Signed-off-by: Andrew Kofink <akofink@redhat.com>

### CODE REVIEW

**Comment by akofink** [comment]
@dLobatog @bastilian this one should be good to go now

File: insights/client/apps/compliance/__init__.py [hunk 1]
```diff
@@ -22,9 +23,10 @@ def __init__(self, config):
         self.hostname = get_canonical_facts().get('fqdn', '')
 
     def oscap_scan(self):
+        self._assert_oscap_rpms_exist()
         policies = self.get_policies()
         if not policies:
-            logger.error("ERROR: System does not exist!\n")
+            logger.error("System is not associated with any profiles. Assign profiles by either uploading a SCAP scan or using the compliance web UI.\n")
```
Comment (akofink) [inline insights/client/apps/compliance/__init__.py]: @katierik UX check on this error message. This occurs when the system is registered with insights, but there are no associated profiles in the compliance app.

Comment (katierik) [reply insights/client/apps/compliance/__init__.py]: This is a huge improvement. What did the user do to get here?

Comment (akofink) [reply insights/client/apps/compliance/__init__.py]: This is one of the failure modes of running `insights-client --compliance`. In order for the client to run a SCAP scan against a policy, the client needs to query the compliance app for associated policies for this host. If there are no policies associated with this host, this is the error message displayed. Feel free to reach out to me if you still have questions.

**Comment by akofink** [comment]
Tests passed (finally)

**Comment by akofink** [comment]
@RedHatInsights/insights-core-devs this one should be good to go

**Comment by akofink** [comment]
@gravitypriest what do you think?

**Comment by gravitypriest** [comment]
Looks good to me. Will merge when checks pass.

**Pull request merged**